File: python/examples/01_authenticate.py
"""Confirm the credential works and print what the token can actually do.

Run this first when an integration misbehaves: it separates a configuration problem (wrong
company, missing scope, feature disabled) from a bug in your code.
"""

import sys

from salesbud import SalesbudClient, SalesbudError

client = SalesbudClient.from_env()

try:
    data = client.context()["data"]
except SalesbudError as error:
    print(f"Failed: {error.code} (HTTP {error.status})", file=sys.stderr)
    print(f"  {error}", file=sys.stderr)
    if error.request_id:
        print(f"  request_id {error.request_id}", file=sys.stderr)

    # These two are configuration, not transient. Retrying will not help.
    if error.code == "API_ACCESS_DISABLED":
        print("\n  Your company does not have API access enabled.", file=sys.stderr)
    if error.code == "invalid_client":
        print("\n  Wrong client id or secret, or the credential was revoked.", file=sys.stderr)
    raise SystemExit(1)

print("Authenticated.\n")
print(f"  company   {data['company']['name']} ({data['company']['id']})")
print(f"  client    {data['client']['name']}")
print(f"  scopes    {', '.join(data['scopes'])}")
print(f"  quota     {data['rate_limit']['requests_per_minute']} requests / minute")

if "transcriptions.read" not in data["scopes"]:
    print("\n  Note: this credential cannot read transcripts.")

File: node/examples/01-authenticate.js
/**
 * Confirms the credential works and prints what the token can actually do.
 *
 * Run this first when an integration misbehaves: it separates a configuration problem (wrong
 * company, missing scope, feature disabled) from a bug in your code.
 */
import { clientFromEnv, SalesbudError } from "../src/salesbud-client.js";

const client = clientFromEnv();

try {
  const { data } = await client.context();

  console.log("Authenticated.\n");
  console.log(`  company   ${data.company.name} (${data.company.id})`);
  console.log(`  client    ${data.client.name}`);
  console.log(`  scopes    ${data.scopes.join(", ")}`);
  console.log(`  quota     ${data.rate_limit.requests_per_minute} requests / minute`);

  if (!data.scopes.includes("transcriptions.read")) {
    console.log("\n  Note: this credential cannot read transcripts.");
  }
} catch (error) {
  if (!(error instanceof SalesbudError)) throw error;

  console.error(`Failed: ${error.code} (HTTP ${error.status})`);
  console.error(`  ${error.message}`);
  if (error.requestId) console.error(`  request_id ${error.requestId}`);

  // These three are configuration, not transient. Retrying will not help.
  if (error.code === "API_ACCESS_DISABLED") {
    console.error("\n  Your company does not have API access enabled. Talk to your Salesbud contact.");
  }
  if (error.code === "invalid_client") {
    console.error("\n  Wrong client id or secret, or the credential was revoked.");
  }
  process.exit(1);
}
